### PR TITLE
[UNTIL-10894]Scan SDK tests

### PR DESCRIPTION
**Pull request by shekar-allam**

### CODE REVIEW

File: scan-engine/src/test/java/de/tillhub/scanengine/data/ScannerManufacturerTest.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+@RobolectricTest
+internal class ScannerManufacturerTest : FunSpec({
+
+    test("SUNMI device") {
+        ShadowBuild.setManufacturer("SUNMI")
+        ScannerManufacturer.get() shouldBe ScannerManufacturer.SUNMI
+    }
+
+    test("OTHER device") {
+        ShadowBuild.setManufacturer("OTHER")
```
Comment (SloInfinity) [inline scan-engine/src/test/java/de/tillhub/scanengine/data/ScannerManufacturerTest.kt:17]: based on the code we want to test with something other than sunmi, without context I'd assume that "OTHER" has a proper mapping. 
I am suggesting to change "OTHER" to "NOT_SUNMI"

**Review by SloInfinity — APPROVED** [review]

File: scan-engine/src/androidTest/java/de/tillhub/scanengine/barcode/SunmiBarcodeScannerTest.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    fun setUp() {
+        context = ApplicationProvider.getApplicationContext()
+        mutableScanEvents = MutableSharedFlow(replay = 1)
+        SunmiBarcodeScanner(context, mutableScanEvents)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun testBroadcastReceiverRegistered() {
+        val intent = Intent(SunmiBarcodeScanner.ACTION_DATA_CODE_RECEIVED).apply {
```
Comment (djordjeh) [inline scan-engine/src/androidTest/java/de/tillhub/scanengine/barcode/SunmiBarcodeScannerTest.kt:40]: In tests use hard coded string here "com.sunmi.scanner.ACTION_DATA_CODE_RECEIVED" we don't want to use variable from actual code.

File: scan-engine/src/main/java/de/tillhub/scanengine/barcode/SunmiBarcodeScanner.kt [hunk 3]
```diff
@@ -40,7 +40,7 @@ internal class SunmiBarcodeScanner(
     }
 
     companion object {
-        private const val ACTION_DATA_CODE_RECEIVED = "com.sunmi.scanner.ACTION_DATA_CODE_RECEIVED"
+        internal const val ACTION_DATA_CODE_RECEIVED = "com.sunmi.scanner.ACTION_DATA_CODE_RECEIVED"
         const val DATA = "data"
```
Comment (djordjeh) [inline scan-engine/src/main/java/de/tillhub/scanengine/barcode/SunmiBarcodeScanner.kt:44]: DATA can be private as well.

**Review by djordjeh — CHANGES_REQUESTED** [review]

File: scan-engine/src/main/java/de/tillhub/scanengine/google/ui/GoogleScanningActivity.kt [hunk 4]
```diff
@@ -34,25 +35,29 @@ internal class GoogleScanningActivity : AppCompatActivity() {
...
+            }
         }
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         setContentView(binding.root)
+        setSupportActionBar(binding.toolbar)
+        supportActionBar?.title = EMPTY
```
Comment (djordjeh) [inline scan-engine/src/main/java/de/tillhub/scanengine/google/ui/GoogleScanningActivity.kt:60]: There is better way supportActionBar.setDisplayShowTitleEnabled(false) and you can delete EMPTY

**Review by djordjeh — CHANGES_REQUESTED** [review]

**Review by djordjeh — COMMENTED** [review]
Since it will be some refactoring with additional bluetooth scanners, let merge this. After you finish refactoring tests will be addressed according to new code.

**Review by djordjeh — APPROVED** [review]

**Pull request merged**